hudi-hadoop-common/src/main/java/org/apache/hudi/common/util/HFileUtils.java

```diff
@@ -207,25 +194,21 @@ public byte[] serializeRecordsToLogBlock(HoodieStorage storage,
       sortedRecordsMap.put(recordKey, recordBytesList);
     }
 
-    HFile.Writer writer = HFile.getWriterFactory(conf, cacheConfig)
-        .withOutputStream(ostream).withFileContext(context).create();
-
-    // Write the records
+    HFileContext context = HFileContext.builder().build();
+    HFileWriter writer = new HFileWriterImpl(context, ostream);
     sortedRecordsMap.forEach((recordKey, recordBytesList) -> {
       for (byte[] recordBytes : recordBytesList) {
         try {
-          KeyValue kv = new KeyValue(recordKey.getBytes(), null, null, recordBytes);
-          writer.append(kv);
+          writer.append(recordKey.getBytes(StandardCharsets.UTF_8), recordBytes);
```

Comment:
Let's remove all Hadoop class usage in this class and move this `HFileUtils` class to `hudi-common`.